src/GafferScene/Instancer.cpp

```diff
@@ -2528,82 +2529,112 @@ void Instancer::PrototypeScope::setPrototype( const EngineData *engine, const Sc
 namespace
 {
 
-struct Prototype
+// It shouldn't be necessary for this to be refcounted - but LRUCache is set up to make it impossible
```

Comment:
I ( thought I ) had some spare time today, so I took a swing at this. It resulted in a 20% speedup in testInstancerManyPrototypesEncapsulatePerf ... this surprised me a bit - I had expected it to be more significant in testInstancerFewPrototypesEncapsulatePerf, since I was expecting the overhead in my code to play a greater part when the LRUCache just has one thing in it, and isn't working that hard. These results suggest that there's some overhead inside the LRUCache ... are we doing reference counting when the hash map resizes? I would have hoped that modern C++ could prevent that with an `std::move`? Anyway, that might merit some investigation - but I think this faster code feels a bit cleaner to me anyway? Happy to omit this commit if it raises some uncertainty for you though that would prevent merging this PR.

Reply:
Not sure if this explains what you're seeing, but I'd expect the LRUCache to perform better when queries are spread across a number of different keys. Internally it uses a binned map and mutex on each entry, so it sees much less contention if different threads are asking for different things than if they're all fighting over the one thing. Perhaps that fighting-over-one-thing drowns out any potential benefit from removing the ref-counting?